src/middlewares/authentication/AuthenticationMiddleware.ts

```diff
@@ -0,0 +1,33 @@
+import { ExpressMiddlewareInterface } from 'routing-controllers';
+import { singleton, inject } from 'tsyringe';
+
+import { AuthenticationService } from '@Services';
+import { config } from '../../config';
+import * as ERR_MSGS from '../../constants/ErrResponses';
+
+@singleton()
+export class AuthenticationMiddleware implements ExpressMiddlewareInterface {
+  private authenticationService: AuthenticationService;
+
+  constructor(@inject(AuthenticationService) authenticationService: AuthenticationService) {
+    this.authenticationService = authenticationService;
+  }
+
+  async use(request: any, response: any, next?: (err?: any) => any): Promise<any> {
+    const { headers } = request;
+    const token = headers['authorization'];
+    const { customEnvMode } = config;
+
+    try {
+      const tokenResult = await this.authenticationService.verifyToken(customEnvMode, token);
+
+      if (tokenResult !== undefined) {
```

Comment:
can probably drop the else and do if (tokenResult === undef) {return next(blah)} return next()